File: lib/rest/yampi/mappers/yampi_address_mapper.dart
import 'package:sertton/core/checkout/dtos/address_dto.dart';
import 'package:sertton/rest/types/json.dart';

class YampiAddressMapper {
  static AddressDto toDto(Json json) {
    if (json.containsKey('data')) {
      json = json['data'];
    }

    return AddressDto(
      id: json['id']?.toString() ?? '',
      receiver: json['receiver'] ?? '',
      zipcode: json['zipcode'] ?? '',
      street: json['street'] ?? '',
      number: json['number'] ?? '',
      neighborhood: json['neighborhood'] ?? '',
      complement: json['complement'] ?? '',
      city: json['city'] ?? '',
      uf: json['uf'] ?? json['state'] ?? '',
    );
  }
}

File: src/rest/yampi/mappers/yampi-address-mapper.ts
import type { Address } from "@/core/checkout/entities"
import type { YampiAddress } from "@/rest/yampi/types"

export const YampiAddressMapper = () => ({
  toDomain(input: YampiAddress): Address {
    return {
      id: String(input.id),
      receiver: input.receiver,
      zipcode: String(input.zipcode),
      street: input.street,
      number: String(input.number),
      neighborhood: input.neighborhood,
      complement: input.complement ?? "",
      city: input.city,
      uf: input.uf || input.state,
    }
  },
})
